httpserver/httpserver.go

```diff
@@ -23,22 +24,27 @@ type Server struct {
 
 type ServerOption func(*Server)
 
-// WithHandler adds a handler on the http endpoint.
-func WithHandler(h HandleChainFunc) ServerOption {
+// WithHandler adds a handler on the http endpoint. Nil argument is ignored.
+func WithHandler(handler HandleChainFunc) ServerOption {
 	return func(s *Server) {
-		if h != nil {
-			s.handlers = append(s.handlers, h)
+		if handler != nil && !reflect.ValueOf(handler).IsNil() {
+			s.handlers = append(s.handlers, handler)
 		}
 	}
 }
 
-// WithHandlers adds a handler on the http endpoint.
-func WithHandlers(h []HandleChainFunc) ServerOption {
+// WithHandlers adds a handler on the http endpoint. Nil arguments is ignored.
+func WithHandlers(handlers []HandleChainFunc) ServerOption {
```

Comment:
Instead of having two function signatures that are the same, one having a singular `HandleChainFunc` the other having many, can we not just do `func WithHandlers(handlers ...HandleChainFunc)`?

Reply:
good idea

Reply:
fixed